session.go

```diff
@@ -1537,8 +1537,21 @@ func simpleIndexKey(realKey bson.D) (key []string) {
 		field := realKey[i].Name
 		vi, ok := realKey[i].Value.(int)
 		if !ok {
-			vf, _ := realKey[i].Value.(float64)
-			vi = int(vf)
+			switch realKey[i].Value.(type){
+			case int64:
+				vf, _ := realKey[i].Value.(int64)
+				vi = int(vf)
+			case float64:
+				vf, _ := realKey[i].Value.(float64)
+				vi = int(vf)
+			case string:
+				if vs, ok := realKey[i].Value.(string); ok {
+					key = append(key, "$"+vs+":"+field)
+					continue
+				}
+			default:
+				panic("Got unknown index key type for field " + field)
```

Comment:
I've removed it, but there is a similar panic right below it that was the reason I included it.